server/worker/apple_mdm.go

```diff
@@ -150,19 +159,50 @@ func (a *AppleMDM) runPostDEPEnrollment(ctx context.Context, args appleMDMArgs)
 			); err != nil {
 				return ctxerr.Wrap(ctx, err, "sending AccountConfiguration command")
 			}
+			awaitCmdUUIDs = append(awaitCmdUUIDs, cmdUUID)
+		}
+	}
+
+	// proceed to release the device only if it is not a macos, as those are
+	// released via the setup experience flow.
+	if !isMacOS(args.Platform) {
+		var manualRelease bool
+		if args.TeamID == nil {
+			ac, err := a.Datastore.AppConfig(ctx)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "get AppConfig to read enable_release_device_manually")
+			}
+			manualRelease = ac.MDM.MacOSSetup.EnableReleaseDeviceManually.Value
+		} else {
+			tm, err := a.Datastore.Team(ctx, *args.TeamID)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "get Team to read enable_release_device_manually")
+			}
+			manualRelease = tm.Config.MDM.MacOSSetup.EnableReleaseDeviceManually.Value
+		}
+
+		if !manualRelease {
+			// send all command uuids for the commands sent here during post-DEP
+			// enrollment and enqueue a job to look for the status of those commands to
+			// be final and same for MDM profiles of that host; it means the DEP
+			// enrollment process is done and the device can be released.
+			if err := QueueAppleMDMJob(ctx, a.Datastore, a.Log, AppleMDMPostDEPReleaseDeviceTask,
+				args.HostUUID, args.Platform, args.TeamID, args.EnrollReference, awaitCmdUUIDs...); err != nil {
+				return ctxerr.Wrap(ctx, err, "queue Apple Post-DEP release device job")
+			}
 		}
 	}
 
 	return nil
 }
 
-// This job is deprecated because releasing devices is now done via the orbit
-// endpoint /setup_experience/status that is polled by a swift dialog UI window
-// during the setup process, and automatically releases the device once all
-// pending setup tasks are done. However, it must remain implemented in case
-// there are such jobs to process after a Fleet migration to a new version; we
-// just don't enqueue that job anymore.
-func (a *AppleMDM) deprecatedRunPostDEPReleaseDevice(ctx context.Context, args appleMDMArgs) error {
+// This job is deprecated for macos because releasing devices is now done via
+// the orbit endpoint /setup_experience/status that is polled by a swift dialog
+// UI window during the setup process, and automatically releases the device
+// once all pending setup tasks are done. However, it must remain implemented
+// for iOS and iPadOS and in case there are such jobs to process after a Fleet
+// migration to a new version.
+func (a *AppleMDM) runPostDEPReleaseDevice(ctx context.Context, args appleMDMArgs) error {
```

Comment:
I have a failed attempt at covering the automatic release flow for an ios device (the commit I reverted in this PR), but unfortunately I think it would require a bit more work and more knowledge of how the iDevices enroll - the current test MDM client seems to only support macs for DEP: https://github.com/fleetdm/fleet/blob/e2d9a9016cb2bbf6cfe0ef7512f58d9934bf69de/pkg/mdm/mdmtest/apple.go#L128-L138

So as it stands, there is the existing tests that verifies that the worker job is not enqueued anymore for macs, and we'll have to rely on QA to confirm the iDevice fix.